test/acquisition/multi_objective/test_monte_carlo.py

```diff
@@ -1169,6 +1271,38 @@ def test_constrained_q_noisy_expected_hypervolume_improvement(self):
             mm._posterior._samples = samples
             res = acqf(X)
             self.assertAlmostEqual(res.item(), 1.5, places=4)
+            # test multiple constraints one eta with
+            # this crashes for large etas, and I do not why
+            # set the MockPosterior to use samples over baseline points
+            mm._posterior._samples = baseline_samples
+            sampler = IIDNormalSampler(sample_shape=torch.Size([1]))
+            acqf = qNoisyExpectedHypervolumeImprovement(
+                model=mm,
+                ref_point=ref_point,
+                X_baseline=X_baseline,
+                sampler=sampler,
+                constraints=[
+                    # lambda Z: torch.zeros_like(Z[..., -1]),
+                    lambda Z: torch.ones_like(Z[..., -1]),
+                ],
+                eta=1,
+                cache_root=False,
+            )
+            samples = torch.cat(
+                [
+                    baseline_samples.unsqueeze(0),
+                    torch.tensor([[[6.5, 4.5]]], **tkwargs),
+                ],
+                dim=1,
+            )
+            mm._posterior._samples = samples
```

Comment:
Re test failure: It seems like you expect the infeasible acqf value to be the feasible acqf value times the sigmoid value of the constraints there. This does not work there, since the two acqfs compute the hypervolume improvement w.r.t two different partitionings. The feasible acqf has a partitioning that has a non-empty set of baseline points (you can print `acqf.partitioning.Y`) but the infeasible acqf has a partitioning that is empty. This difference happens because the constraints are taken into account while constructing the partitioning in `__init__ -> _set_cell_bounds`. 

If instead of constructing a new acqf in here, you just update the constraints and eta as below, the test will pass.
```suggestion
            acqf.constraints = [
                # lambda Z: torch.zeros_like(Z[..., -1]),
                lambda Z: torch.ones_like(Z[..., -1]),
            ]
            acqf.eta = torch.tensor([1.0])
```

Reply:
Thank you very much for the explanation! I updated the tests accordingly.